### DISCUSSION TITLE
React Vue or Angular?

### DISCUSSION THREAD

**Discussion by AshNiz24** [community/community#38255] [Other Feature Feedback, Questions, & Ideas] [answered]
Which is better React Vue or Angular

**Comment by chandran-jr** [comment, marked as answer]
Angular is a framework by Google.
Over the years it lostr popularity due to various reasons like it being heavy and
not being able to implement a lot of functionalities although it is used by many companies still.

React, my personal favourite was built by Meta (Facebook) which is also a JS framework and is very easy to learn,
implement, deploy and is light weight as well with lot of functionalities and hooks available for use.
It is the most popular by a long way right now as seen on stack overflow.

Vue is a relatively new framework which is said to be easier than React but is not as popular. Probabbly might gain
traction after a few years.

Personal opinion: React is used by a lot of companies and can even learn React native for mobile app development.
It is always an in-demand skill. So I'd suggest use React.

**Comment by RafaelNevesdeOliveira** [comment]
Depends, to me the best frameword is that one pay you.
As @chandran-jr said above, React is easiear but personally I prefer Angular due structure and tools that it provide.

**Comment by AliRazaSaqib** [comment]
React, Vue, and Angular are all popular and powerful JavaScript frameworks for building web applications, and each has its own strengths and weaknesses.

React is known for its flexibility and scalability, and is often used for building complex user interfaces. It is backed by Facebook and has a large and active community, which means that there are many resources and libraries available to help developers work with it.

Vue is a progressive framework that is designed to be easy to learn and use, with a relatively low barrier to entry. It is often compared to React, but has some key differences, including a simpler syntax and a more lightweight approach to building applications. Vue is also known for its performance, and is a good choice for smaller projects or for building parts of a larger application.

Angular, on the other hand, is a full-featured framework that is often used for building large-scale enterprise applications. It has a steep learning curve, but offers many powerful features and tools for building complex applications. Angular also includes features such as dependency injection and testing tools, which can be useful for developers working on large projects.

Ultimately, the choice of which framework to use will depend on the specific needs of your project and your own personal preferences and experience as a developer. All three frameworks are powerful tools that can help you build robust, high-quality web applications.

**Comment by canyon920** [comment]
React, and Vue js  popular javascript SPA framework, 
React using many peopl bedause,  React proves its performance by being the most popular framework at the moment.
Angular has a long history, so there are many things to study, but these days, React and Vue are convenient, so people don't use it often
